File: solution/1200-1299/1290.Convert Binary Number in a Linked List to Integer/Solution.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func getDecimalValue(head *ListNode) (ans int) {
	for ; head != nil; head = head.Next {
		ans = ans<<1 | head.Val
	}
	return
}

File: solution/1200-1299/1290.Convert Binary Number in a Linked List to Integer/Solution.php
/**
 * Definition for a singly-linked list.
 * class ListNode {
 *     public $val = 0;
 *     public $next = null;
 *     function __construct($val = 0, $next = null) {
 *         $this->val = $val;
 *         $this->next = $next;
 *     }
 * }
 */
class Solution {
    /**
     * @param ListNode $head
     * @return Integer
     */
    function getDecimalValue($head) {
        $ans = 0;
        while ($head !== null) {
            $ans = ($ans << 1) | $head->val;
            $head = $head->next;
        }
        return $ans;
    }
}